src/Kiota.Builder/Refiners/DartRefiner.cs

```diff
@@ -0,0 +1,501 @@
+                ],
+                static s => s.ToCamelCase(UnderscoreArray));
+
+            AddQueryParameterExtractorMethod(generatedCode);
+            // This adds the BaseRequestBuilder class as a superclass
+            MoveRequestBuilderPropertiesToBaseType(generatedCode,
+                new CodeUsing
+                {
+                    Name = "BaseRequestBuilder",
+                    Declaration = new CodeType
+                    {
+                        Name = AbstractionsNamespaceName,
+                        IsExternal = true,
+                    }
+                }, addCurrentTypeAsGenericTypeParameter: true);
+            RemoveRequestConfigurationClasses(generatedCode,
+                new CodeUsing
+                {
+                    Name = "RequestConfiguration",
+                    Declaration = new CodeType
+                    {
+                        Name = AbstractionsNamespaceName,
+                        IsExternal = true
+                    }
+                }, new CodeType
+                {
+                    Name = "DefaultQueryParameters",
+                    IsExternal = true,
+                });
+            MoveQueryParameterClass(generatedCode);
+            AddDefaultImports(generatedCode, defaultUsingEvaluators);
+            AddPropertiesAndMethodTypesImports(generatedCode, true, true, true, codeTypeFilter);
+            AddParsableImplementsForModelClasses(generatedCode, "Parsable");
+            AddConstructorsForDefaultValues(generatedCode, true);
+            AddConstructorForErrorClass(generatedCode);
+            cancellationToken.ThrowIfCancellationRequested();
+            AddAsyncSuffix(generatedCode);
+            AddDiscriminatorMappingsUsingsToParentClasses(generatedCode, "ParseNode", addUsings: true, includeParentNamespace: true);
+
+            ReplaceReservedNames(generatedCode, reservedNamesProvider, x => $"{x}Escaped");
+            ReplaceReservedModelTypes(generatedCode, reservedNamesProvider, x => $"{x}Object");
+            ReplaceReservedExceptionPropertyNames(
+                generatedCode,
+                new DartExceptionsReservedNamesProvider(),
+                static x => $"{x.ToFirstCharacterLowerCase()}Escaped"
+            );
+
+            ReplaceDefaultSerializationModules(
+                generatedCode,
+                defaultConfiguration.Serializers,
+                new(StringComparer.OrdinalIgnoreCase) {
+                    $"{SerializationNamespaceName}_json/{SerializationNamespaceName}_json.JsonSerializationWriterFactory",
+                    $"{SerializationNamespaceName}_text/{SerializationNamespaceName}_text.TextSerializationWriterFactory",
+                    $"{SerializationNamespaceName}_form/{SerializationNamespaceName}_form.FormSerializationWriterFactory",
+                    $"{SerializationNamespaceName}_multi/{SerializationNamespaceName}_multi.MultipartSerializationWriterFactory",
```

Comment:
I think this should read:

```suggestion
                    $"{SerializationNamespaceName}_multipart/{SerializationNamespaceName}_multipart.MultipartSerializationWriterFactory",
```